scarb/src/core/lockfile.rs

```diff
@@ -184,6 +184,25 @@ impl Lockfile {
         Ok(doc)
     }
 
+    /// Creates a set of [`PackageId`]s present in Lockfile.
+    pub fn create_yanked_whitelist(&self) -> HashSet<PackageId> {
+        self.packages
+            .values()
+            .flatten()
+            .filter_map(|p| {
+                if p.source.is_some() {
```

Comment:
nit If this method is specialized for yanked whitelist, we could even filter to only keep reigstry sources, as yanking has no meaning for git ones (and paths are dropped already) 🤔